### ISSUE TITLE
git status: modified files from git lfs, inconsistent

### ISSUE THREAD

**Issue by guillaume-chevalier** [git-lfs/git-lfs#2325]
Similar to:
https://github.com/git-lfs/git-lfs/issues/1339

To reproduce: 

1. `git clone` via ssh. After that, the git lfs files are dummy lightweight pointers.
2. `git lfs pull` pulls the files as expected, making them huge again. 
3. `git status` now shows the files pulled by git lfs as `modified` AND those files are already added like if I did `git add` them whilst ignoring `git lfs`. 

`git lfs` is full of bugs and it is getting pretty annoying. I hope this is not a feature.

**Comment by ttaylorr** [comment]
@guillaume-chevalier thanks for opening this issue, and sorry that you're having trouble. I think the issue might be that there are large files committed into your history (without LFS), and LFS is trying to convert them on checkout, causing the files to appear as modified.

Can you run `git lfs status` and post the output here to confirm?

**Comment by guillaume-chevalier** [comment]
Here is an output example: 

```
Git LFS objects to be committed:

	large_file.hdf5 (LFS: b850d31 -> Git: b850d31)

Git LFS objects not staged for commit:

	large_file.hdf5 (LFS: b850d31 -> Git: b850d31)
```

And here if I try to git add another file:

```
Git LFS objects to be committed:

	large_file.hdf5 (LFS: b850d31 -> Git: b850d31)
	other_large_file.hdf5 (Git: 290ec4d -> File: 290ec4d)

Git LFS objects not staged for commit:

	large_file.hdf5 (LFS: b850d31 -> Git: b850d31)
```

And here is `cat .gitattributes`:
```
*.{jpg,hdf5} filter=lfs diff=lfs merge=lfs -text
```

**Comment by guillaume-chevalier** [comment]
@ttaylorr So, in fact, the files seems to be already under the LFs format and it want to convert them to Git. I already ran BGF successfully. I have spent a full day on this and yet it is sad. Do you have any idea of what's going on? I am stacking workaround after workaround. 

Note: adding `*.{jpg,hdf5}` to the `.gitignore` changes nothing to the git status nor the git lfs status: the files are still added like with `git add -f` when `git lfs pull` runs. 
ALSO: `git status` takes a lot of time to run after the `git lfs pull`, but was running fast before that. 
ALSO: I have verified and I confirm that my bitbucket repo is not heavy, the huge files are correctly pushed only to the bitbucket LFS part.

**Comment by ttaylorr** [comment]
@guillaume-chevalier Taking a look at your `git-lfs-status(1)` output, I'm curious about why your repository is converting LFS tracked files to Git ones:

> ```
> Git LFS objects to be committed:
>
> 	large_file.hdf5 (LFS: b850d31 -> Git: b850d31)
> ```

The above means that the `large_file.hdf5` entry is stored with LFS in your repository's history, but is being staged without LFS (albeit using the same content hash). Do you have the LFS pre-push hook installed? You can install it via `git lfs install --force`.

**Comment by doxxx** [comment]
@guillaume-chevalier 

I think the problem here is that `.gitattributes` does not understand bash-style brace expansion patterns. If you separate that pattern into two lines, it should work:

*.jpg filter=lfs diff=lfs merge=lfs -text
*.hdf5 filter=lfs diff=lfs merge=lfs -text

**Comment by bgurmendi** [comment]
Similar issue hier

```
> git lfs status
On branch develop

Git LFS objects to be committed:

	web/xxxx/xxx/xxxxx.odt (Git: 99891d1 -> LFS: 99891d1)

Git LFS objects not staged for commit:
```

I'm prety sure that whe have a commited largue file without lfs pointer in the history. 

But in this case, I would expect git-lfs to ignore the file, perhaps with a warning with the hash of the confirmation and the name of the developer, to blame him.

The explanation is: someone has already committed a file to the history so there is no easy solution to this situation, git-lfs could report about the problem but not interrupt the work of other developers.

**Comment by ttaylorr** [comment]
> so there is no easy solution to this situation, git-lfs could report about the problem but not interrupt the work of other developers.

You're absolutely correct in noting that this is a difficult case for Git LFS to handle correctly. If someone commits a large object in place of a pointer file, copying the file over verbatim as a large blob will leave the working copy as modified, because Git expects an LFS pointer file to be there (without having run the smudge filter).

So, the solution is to rewrite the affected portions of history by using the `git lfs migrate` command, which make it possible to "migrate" broken commits containing large objects to be using Git LFs.
